core/domain/rights_manager.py

```diff
@@ -945,6 +1076,10 @@ def _deassign_role(committer, removed_user_id, activity_id, activity_type):
                 committer_id, removed_user_id, activity_id))
         raise Exception(
             'UnauthorizedUserException: Could not deassign role.')
+
+    # Ruling out the possibility of None for mypy type checking.
```

Comment:
Ditto.

Reply:
Similarly, in line 1085, we are assessing `owner_ids` without checking the none case of `activity_rights`.

Reply:
See comments above.

Reply:
Similarly, in line `1091` we are already checking if activity_rights is None or not.

In `check_can_modify_core_activity_roles()` method, 
https://github.com/oppia/oppia/blob/0e69c2bcd7a91ef3595da274d060608efd3140ae/core/domain/rights_manager.py#L938

If we provide activity_rights as None then it going to return a False value and that False value is assigned to `user_can_deassign_role`. So, if `user_can_deassign_role` is False then an error is raised in line 1094.

Reply:
I have the same comment as above. In general I suggest that for the purpose of deciding whether or not adding assertions is safe, you don't rely on the behaviour of other called functions, because those may change.

Reply:
if clause is added and docstring is updated.